lib/rules/no-misleading-character-class.js

```diff
@@ -197,32 +233,39 @@ module.exports = {
 
         /**
          * Generates a granular loc for context.report.
-         * @param {Character} char Individual character being reported on.
-         * @param {string} kind What kind of character report is being made.
+         * @param {Character[]} chars Individual characters being reported on.
          * @param {Node} node Parent string node to report within.
          * @returns {Object} Granular loc for context.report.
-         * @see https://github.com/eslint/eslint/pull/17515#issuecomment-1707697186
+         * @see https://github.com/eslint/eslint/pull/17515
          */
-        function generateReportLocation(char, kind, node) {
-            if (node.type === "TemplateLiteral" && node.expressions.length || sourceCode.getText(node).includes("\\")) {
+        function generateReportLocation(chars, node) {
+            if (node.type !== "Literal" && (node.type !== "TemplateLiteral" || node.expressions.length)) {
+                return node.loc;
+            }
+
+            const first = chars[0];
+            const last = chars[chars.length - 1];
+
+            const reportStartOffset = countDoubleEscapedBackslashes(node.raw.slice(0, first.start));
```

Comment:
It would be better to use `sourceCode.getText(node)` instead of `node.raw` for best compatibility with all parsers. The `raw` property is not part of ESTree, so we can't know if all parsers will return it or not.